### PR TITLE
[testing] avoid using werf inside the docker

### PR DESCRIPTION
**Pull request by kvaps**
## Description

Using werf insude the docker container brings additional problems:
- Werf creates new git objects with wrong permissions, which must be fixed every time after running docker
- Werf wants to control git worktrees, which are persistent and can be broken insude/outside container

## Why do we need it, and what problem does it solve?

It seems the [previous fix](https://github.com/deckhouse/deckhouse/pull/2848) does not working properly
This is another attempt to fix it.

## What is the expected result?

No more werf inside the docker, let it on a host only

## Checklist
- [ ] The code is covered by unit tests.
- [ ] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [ ] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: testing
type: fix
summary: avoid using werf inside the docker
impact: low
```

### CODE REVIEW

**Comment by kvaps** [comment]
Tests success: https://github.com/deckhouse/deckhouse-test-1/actions/runs/3420445788

File: .github/ci_templates/build.yml [hunk 1]
```diff
@@ -5,12 +5,16 @@ steps:
...
   {!{ tmpl.Exec "login_readonly_registry_step" . | strings.Indent 2 }!}
+  {!{ tmpl.Exec "werf_install_step"            . | strings.Indent 2 }!}
 
   - name: Run go generate
     run: |
       TESTS_IMAGE_NAME={!{ printf "%s%s" (datasource "image_versions").REGISTRY_PATH (datasource "image_versions").BASE_GOLANG_18_BUSTER | quote }!}
-      docker run -v $(pwd):/deckhouse --tmpfs /deckhouse/.git/worktrees -v ~/go-pkg-cache:/go/pkg -v ~/deckhouse-bin-cache:/deckhouse/bin -w /deckhouse ${TESTS_IMAGE_NAME} sh -c "trap 'chown -R $(id -u):$(id -g) /deckhouse' EXIT; make generate"
-      docker run -v $(pwd):/deckhouse --tmpfs /deckhouse/.git/worktrees -v ~/go-pkg-cache:/go/pkg -v ~/deckhouse-bin-cache:/deckhouse/bin -w /deckhouse/modules/500-upmeter/hooks/smokemini/internal/snapshot ${TESTS_IMAGE_NAME} go generate .
+      RENDERED_WERF_CONFIG="$(mktemp)"
+      CI_COMMIT_REF_NAME= CI_COMMIT_TAG= WERF_ENV=FE werf config render --dev --log-quiet > "$RENDERED_WERF_CONFIG"
```
Comment (diafour) [inline .github/ci_templates/build.yml:14]: There are no unstaged changes, is `--dev` needed here?

**Comment by kvaps** [comment]
> Why shell-operator/frameworks/shell/.gitkeep ?
Sry, fixed

**Comment by kvaps** [comment]
@diafour suggested to get rid of docker in this step fully.
PR updated. Pipeline succeeded:
https://github.com/deckhouse/deckhouse-test-1/actions/runs/3427513480

File: Makefile [hunk 2]
```diff
@@ -224,15 +220,6 @@ bin/yq: bin ## Install yq deps for update-patchversion script.
 bin/crane: bin ## Install crane deps for update-patchversion script.
 	curl -sSfL https://github.com/google/go-containerregistry/releases/download/v0.10.0/go-containerregistry_$(OS_NAME)_$(CRANE_ARCH).tar.gz | tar -xzf - crane && mv crane bin/crane && chmod +x bin/crane
 
-bin/trdl: bin
-	curl -sSfL https://tuf.trdl.dev/targets/releases/0.6.3/$(TRDL_PLATFORM)-$(TRDL_ARCH)/bin/trdl -o bin/trdl
-	chmod +x bin/trdl
-
-bin/werf: bin bin/trdl ## Install werf for images-tags generator.
```
Comment (apolovov) [inline Makefile:231]: Don't remove werf.

Comment (kvaps) [reply Makefile]: done

File: .github/ci_templates/build.yml (hunk 1 above)
Comment (kvaps) [reply .github/ci_templates/build.yml]: this part of code removed in second commit

File: .github/ci_templates/build.yml [hunk 3]
```diff
@@ -5,12 +5,17 @@ steps:
...
+    uses: actions/setup-go@v3
+    with:
+      go-version: 1.16
 
   - name: Run go generate
     run: |
-      TESTS_IMAGE_NAME={!{ printf "%s%s" (datasource "image_versions").REGISTRY_PATH (datasource "image_versions").BASE_GOLANG_18_BUSTER | quote }!}
-      docker run -v $(pwd):/deckhouse --tmpfs /deckhouse/.git/worktrees -v ~/go-pkg-cache:/go/pkg -v ~/deckhouse-bin-cache:/deckhouse/bin -w /deckhouse ${TESTS_IMAGE_NAME} sh -c "trap 'chown -R $(id -u):$(id -g) /deckhouse' EXIT; make generate"
-      docker run -v $(pwd):/deckhouse --tmpfs /deckhouse/.git/worktrees -v ~/go-pkg-cache:/go/pkg -v ~/deckhouse-bin-cache:/deckhouse/bin -w /deckhouse/modules/500-upmeter/hooks/smokemini/internal/snapshot ${TESTS_IMAGE_NAME} go generate .
+      (cd tools && go generate)
```
Comment (diafour) [inline .github/ci_templates/build.yml:17]:
```suggestion
      {!{/* TODO replace with 'make generate' when make become available on runners */}!}
      (cd tools && go generate)
```

**Review by diafour — APPROVED** [review]
lgtm

**Review by apolovov — APPROVED** [review]

**Pull request merged**